### DISCUSSION TITLE
[A-L] (13 Febbraio 2024) Esercizio 1

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#405] [Esami A-L]
### Fila A

Consideriamo le stringhe binarie di lunghezza 8.

1. Quante iniziano con 10? [1.5 punti]

2. Quante iniziano con 1 o finiscono con 01? [1.5 punti]

3. In quante la cifra 1 compare esattamente 5 volte? [2 punti]

### Fila B

Consideriamo le stringhe binarie di lunghezza 10.

1. Quante finiscono con 11? [1.5 punti]

2. Quante iniziano con 01 o finiscono con 10? [1.5 punti]

3. In quante la cifra 0 compare almeno 2 volte? [2 punti]

**Comment by FeddyLix17** [comment]
### Fila A

Considerando una stringa binaria di lunghezza 8 come $\Large X_1X_2X_3X_4X_5X_6X_7X_8$, dove $\Large X \in \\\{0, 1 \\\}$, si avranno $\Large 2^8$ possibili stringhe diverse.

Tra queste

1. $\Large 1 * 1 * 2^6$ inizieranno con 10

2. $\Large (1 * 2^7) + (2^6 * 1 * 1) - (1 * 2^5 * 1 * 1)$ inizieranno con 1 o finiranno con 01 ([principio di inclusione-esclusione](https://en.wikipedia.org/wiki/Inclusion%E2%80%93exclusion_principle) per gestire

    le stringhe che iniziano con 1 e finiscono con 01 conteggiate 2 volte)

3. $\Large \binom{8}{5}$ la cifrà 1 comparirà esattamente 5 volte (è il numero di modi di scegliere 5 posizioni tra le 8 possibili e inserirci la cifra 1,

    inserendo la cifra 0 nelle restanti 3 ogni volta, di fatto  $\Large \binom{8}{5} = \binom{8}{8-5} = \binom{8}{3}$ )

---

### Fila B

Considerando una stringa binaria di lunghezza 10 come $\Large X_1X_2X_3X_4X_5X_6X_7X_8X_9X_{10}$, dove $\Large X \in \\\{0, 1 \\\}$, si avranno $\Large 2^{10}$ possibili stringhe diverse.

Tra queste

1. $\Large 2^8 * 1 * 1$ finiranno con 11

2. $\Large (1 * 1 * 2^8) + (2^8 * 1 * 1) - (1 * 1 * 2^6 * 1 * 1)$ inizieranno con 01 o finiranno con 10 ([principio di inclusione-esclusione](https://en.wikipedia.org/wiki/Inclusion%E2%80%93exclusion_principle) 

    per gestire le stringhe che iniziano con 01 e finiscono con 10 conteggiate 2 volte)

3. $\Large 2^{10} - \binom{10}{10} - \binom{10}{9}$ conterranno almeno 2 zeri (spiegazione  https://github.com/sapienzastudentsnetwork/metodi-matematici-per-l-informatica/discussions/405#discussioncomment-12277938)

**Reply by DavideGalilei** [reply]
Il 3B è sbagliato. Il numero totale di combinazioni è $\large{2^{10}}$, mentre la soluzione scritta è sbagliata, $\large{\binom{10}{2} \cdot 2^8 = 11520}$, che eccede il totale ($1024$).

La soluzione giusta è partizionare:
- Stringhe con nessuno zero: $\large{1}$ ($\text{1111111111}$)
- Stringhe con uno zero: $\large{10}$ ($\text{0111111111}$, $\text{1011111111}$, $\dots$)

$\large{\text{Soluzione} = \text{Totale} - \text{NessunoZero} - \text{UnoZero} = 2^{10} - 1 - 10 = 1013}$

**Reply by DavideGalilei** [reply]
Un altro approccio per pensare alla 3A, è quello di pensare all'anagramma della stringa $\Large{11111000}$, ovvero $\Large{\frac{8!}{5! \cdot 3!} = \binom{8}{3} = \binom{8}{5}}$. Questo perché dobbiamo eliminare dalle permutazioni della stringa i casi dove la stessa cifra è presente in posizioni diverse, ad esempio $1111$ viene contato diversamente da $1111$ come se fossero cifre distinte, ma non è vero.
